File: activity/app/src/main/java/com/rikkei/training/activity/base/BaseActivity.java
package com.rikkei.training.activity.base;

import android.os.Bundle;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(getLayoutId());
        initViews();
        initData();
    }

    protected abstract void initData();

    protected abstract void initViews();

    protected abstract int getLayoutId();

}

File: UIKotlin/app/src/main/java/com/rikkei/training/uikotlin/base/BaseActivity.kt
package com.rikkei.training.uikotlin.base

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(getLayoutId())
        initAct()
    }

    protected abstract fun initAct()
    protected abstract fun getLayoutId(): Int

}
